tests/ImportResolverTest.php

```diff
@@ -51,4 +51,47 @@ public function testAliasedImport()
         $this->assertResolve(Tools::class . '\A', 'Tools\A');
         $this->assertResolve(Tools::class . '\A\B', 'Tools\A\B');
     }
+
+    /**
+     * @expectedException        \InvalidArgumentException
+     * @expectedExceptionMessage Cannot infer the file name from the given ReflectionObject
+     */
+    public function testConstructorWithInvalidInferFileNameShouldThrowInvalidArgumentException()
+    {
+        $resolver = new ImportResolver(new \ReflectionObject(new \Exception));
+    }
+
+    public function testConstructorWithReflectionProperty()
+    {
+        $resolver = new ImportResolver(new \ReflectionProperty(ReflectionTarget::class, 'foo'));
+
+        $this->assertResolve(ReflectionTarget::class, 'ReflectionTarget');
```

Comment:
You're not testing against `$resolver` here, you need to use `assertSame(..., $resolver->resolve(...`